### PR TITLE
Germany_German Support

### PR DESCRIPTION
**Pull request by Luckyz7**
Duplicated changes made for previous support expansions to enable scraping on indeed and monster with ".de" domain. 

-------------------------------------------------------------------------------------------------
There are some problems though:

1. IP ban? 
After a certain threshold of listings is processed (somewhere between 25 and 75), another query would yield this output:
(Note that enabling a vpn allows for further queries)

[2021-01-28 13:55:51,824] [INFO] JobFunnel: Scraping local providers with: ['IndeedScraperGEGer', 'MonsterScraperGEGer']
[2021-01-28 13:55:53,348] [INFO] IndeedScraperGEGer: Found 2 pages of search results for query=python
[2021-01-28 13:55:53,731] [INFO] IndeedScraperGEGer: Scraped 0 job listings from search results pages
[2021-01-28 13:55:53,735] [ERROR] JobFunnel: Failed to scrape jobs for IndeedScraperGEGer
[2021-01-28 13:55:53,737] [INFO] MonsterScraperGEGer: No get() or set() will be done for Job attrs: ['REMOTENESS']
[2021-01-28 13:55:54,605] [ERROR] JobFunnel: Failed to scrape jobs for MonsterScraperGEGer
[2021-01-28 13:55:54,605] [INFO] JobFunnel: Completed all scraping, found 0 new jobs.
[2021-01-28 13:55:54,625] [WARNING] JobFunnel: No new jobs were added to CSV.

2. After the Indeed scraper is done, this error message appears. I've tried multiple province/city configurations. 

C:\Users\Lucky\Scripts>funnel load -s my_settings.yaml
[2021-01-28 13:02:27,031] [INFO] JobFunnel: Scraping local providers with: ['IndeedScraperGEGer', 'MonsterScraperGEGer']
[2021-01-28 13:02:28,423] [INFO] IndeedScraperGEGer: Found 1 pages of search results for query=python
[2021-01-28 13:02:28,987] [INFO] IndeedScraperGEGer: Scraped 23 job listings from search results pages
100%|##################################################################################| 23/23 [00:30<00:00,  1.33s/it]
[2021-01-28 13:02:59,601] [INFO] MonsterScraperGEGer: No get() or set() will be done for Job attrs: ['REMOTENESS']
[2021-01-28 13:03:00,363] [ERROR] JobFunnel: Failed to scrape jobs for MonsterScraperGEGer
Traceback (most recent call last):
  File "C:\Users\Lucky\Scripts\funnel-script.py", line 11, in <module>
    load_entry_point('JobFunnel==3.0.1', 'console_scripts', 'funnel')()
  File "C:\Users\Lucky\AppData\Roaming\Python\Python38\site-packages\jobfunnel\__main__.py", line 28, in main
    job_funnel.run()
  File "C:\Users\Lucky\AppData\Roaming\Python\Python38\site-packages\jobfunnel\backend\jobfunnel.py", line 114, in run
    scraped_jobs_dict = self.scrape()
  File "C:\Users\Lucky\AppData\Roaming\Python\Python38\site-packages\jobfunnel\backend\jobfunnel.py", line 244, in scrape
    self._check_for_inter_scraper_validity(
  File "C:\Users\Lucky\AppData\Roaming\Python\Python38\site-packages\jobfunnel\backend\jobfunnel.py", line 220, in _check_for_inter_scraper_validity
    raise ValueError(
ValueError: Inter-scraper key-id duplicate! 22e7f67c9200c7ce

### CODE REVIEW

**Comment by thebigG** [comment]
Thanks so much for German support!

That issue looks like #123. We are currently working on fixing this. I haven't gotten around to fix it. I'll try my best to fix it in the coming days so you can merge the changes and hopefully then we'll have fully-functiomal German support. 

Really excited about this new addition 👍

**Comment by Luckyz7** [comment]
> 
> 
> Thanks so much for German support!
> 
> That issue looks like #123. We are currently working on fixing this. I haven't gotten around to fix it. I'll try my best to fix it in the coming days so you can merge the changes and hopefully then we'll have fully-functiomal German support.
> 
> Really excited about this new addition 👍

My pleasure, truly. 
I want to say that I'm new to github. This is actually my first commit so don't really know what I'm doing thb. Nonetheless, I'm interested in what the problems are exactly and how they can be/are fixed. Luxembourg support would be really cool as well. The official languages are ENG, GER & FR over there, so if this works out, adding it should be no problem I think. And maybe I can hack that one on my own :)

Would really appreciate it if you could notify me with what you've done when you have it working. 🚀

File: jobfunnel/backend/scrapers/indeed.py [hunk 1]
```diff
@@ -419,4 +420,66 @@ def _get_num_search_result_pages(self, search_url: str, max_pages=0) -> int:
...
         else:
-            return max_pages
\ No newline at end of file
+            return max_pages
+
+
+class IndeedScraperGEGer(BaseIndeedScraper, BaseGEGerScraper):
+     '''Scrapes jobos from indeed.de
+     '''
+     def _get_search_url(self, method: Optional[str] = 'get') -> str:
```
Comment (PaulMcInnis) [inline jobfunnel/backend/scrapers/indeed.py:429]: I think you can remove `_get_search_url` and `_get_num_search_result_pages` here as these are defined by `BaseIndeedScraper`

File: jobfunnel/backend/scrapers/monster.py [hunk 2]
```diff
@@ -406,3 +406,32 @@ def _get_search_url(self, method: Optional[str] = 'get',
...
+        """
+        if method == 'get':
+            return (
+                'https://www.monster.{}/jobs/search/?{}q={}&where={}'
+                    '&rad={}'.format(
+                        self.config.search_config.domain,
+                        f'page={page}&' if page > 1 else '',
+                        self.query,
+                        self.config.search_config.city.replace(' ', '-'),
+                        self._convert_radius(self.config.search_config.radius)
```
Comment (PaulMcInnis) [inline jobfunnel/backend/scrapers/monster.py:431]: Does German's Monster allow searching by state? If so I think perhaps this is missing here.

**Review by PaulMcInnis — CHANGES_REQUESTED** [review]
Great first PR, I have submitted some things for you to review, I think you may be able to have less code in these new classes, as in many cases they seem to be the same as the base class where they are inherited from. 

Once we discuss you can commit any changes, and I'll review again :)

**Comment by PaulMcInnis** [comment]
Closing this since the feature is being provided by PR #136.

**Pull request closed without merging**